src/tox_current_env/hooks4.py

```diff
@@ -0,0 +1,241 @@
+import argparse
+import os
+import platform
+import sys
+import sysconfig
+from pathlib import Path
+from typing import Set
+
+from tox.config.loader.memory import MemoryLoader
+from tox.execute.local_sub_process import (
+    Execute,
+    LocalSubProcessExecuteInstance,
+)
+from tox.plugin import impl
+from tox.tox_env.python.api import PythonInfo
+from tox.tox_env.python.runner import PythonRun
+
+
+@impl
+def tox_register_tox_env(register):
+    register.add_run_env(CurrentEnv)
+    register.add_run_env(PrintEnv)
+
+
+@impl
+def tox_add_option(parser):
+    parser.add_argument(
+        "--current-env",
+        action="store_true",
+        dest="current_env",
+        default=False,
+        help="Run tests in current environment, not creating any virtual environment",
+    )
+    parser.add_argument(
+        "--print-deps-to",
+        "--print-deps-to-file",
+        action="store",
+        type=argparse.FileType("w"),
+        metavar="FILE",
+        default=False,
+        help="Don't run tests, only print the dependencies to the given file "
+        + "(use `-` for stdout)",
+    )
+    parser.add_argument(
+        "--print-extras-to",
+        "--print-extras-to-file",
+        action="store",
+        type=argparse.FileType("w"),
+        metavar="FILE",
+        default=False,
+        help="Don't run tests, only print the  names of the required extras to the given file "
+        + "(use `-` for stdout)",
+    )
+
+
+@impl
+def tox_add_core_config(core_conf, state):
+    opt = state.conf.options
+
+    if opt.current_env or opt.print_deps_to or opt.print_extras_to:
+        # We do not want to install the main package.
+        # no_package is the same as skipsdist.
+        loader = MemoryLoader(no_package=True)
+        core_conf.loaders.insert(0, loader)
+
+    if opt.current_env:
+        opt.default_runner = "current-env"
+        return
+
+    if getattr(opt.print_deps_to, "name", object()) == getattr(
+        opt.print_extras_to, "name", object()
+    ):
+        raise RuntimeError(
+            "The paths given to --print-deps-to and --print-extras-to cannot be identical."
+        )
+
+    if opt.print_deps_to or opt.print_extras_to:
+        opt.default_runner = "print-env"
+        return
+
+    # No options used - switch back to the standard runner
+    # Workaround for: https://github.com/tox-dev/tox/issues/2264
+    opt.default_runner = "virtualenv"
```

Comment:
This has been fixed upstream so can be removed?

Reply:
The issue was closed but I don't see any indication it was fixed. I understood the close more or less as "this is how it works". Did I get that wrong?

Reply:
What about https://github.com/tox-dev/tox/pull/2305 that's linked in the issue?

Reply:
I must have missed that one. We need to check.